doc/type_erasure.md

```diff
@@ -0,0 +1,590 @@
+by looking at the `CPO::type_erased_signature_t` nested type-alias. This will be a function signature
+with exactly one parameter having the `unifex::this_` type as a placeholder representing the type-erased
+object.
+
+A CPO overload is described using one of two approaches:
+* For CPOs that take only one generic argument which is the type-erased wrapper can define
+  a default `type_erased_signature_t` type-alias on the CPO itself.
+* For CPOs that have additional generic parameters, we can use the `unifex::overload()`
+  helper to define a CPO that inherits from the original CPO and adds the
+  `type_erased_signature_t` type-alias for the particular overload being
+  customised.
+
+For example: A CPO that takes a single parameter and always returns a particular type.
+```c++
+struct blocking_t {
+  using type_erased_signature_t = unifex::blocking_kind(const unifex::this_&) noexcept;
+
+  template<typename T>
+    requires unifex::tag_invocable<blocking_t, const T&>
+  unifex::blocking_kind operator()(const T& value) const noexcept {
+    return unifex::tag_invoke(*this, value);
+  }
+};
+
+inline constexpr blocking_t blocking{};
+```
+
+Then the `blocking` CPO can be used directly as a parameter to the type-erased wrapper.
+```c++
+using any_blocking = unifex::any_unique_t<blocking>;
+```
+
+For example: A CPO that has other parameters that need to be specified can use the
+`unifex::overload()` helper.
+```c++
+
+using any_receiver_of_int = unifex::any_unique_t<
+  // Use explicit template parameter to overload()
+  unifex::overload<void(unifex::this_&&, int) noexcept>(unifex::set_value),
+
+  // Use 'sig' helper in conjunction with overload()
+  unifex::overload(unifex::set_error, unifex::sig<void(unifex::this&&, std::exception_ptr) noexcept>)
+  >;
+```
+
+# Type-Erasing Wrapper Types
+
+## `unifex::any_unique`
+
+The `any_unique` type-erasing wrapper is modelled loosely after the ownership semantics of the `std::unique_ptr` smart-pointer.
+
+The `any_unique` type is:
+* *Unique Ownership* - Wrapped object is owned exclusively by the wrapper.
+* *Heap Allocated* - Always stores the wrapped object in a heap-allocation. The allocation may be customised by passing an allocator with `std::allocator_arg` to the constructor.
+* *Hybrid VTable* - Stores VTable inline if 2 or less CPOs listed, otherwise out-of-line (max wrapper size of 4 pointers).
+* *Nothrow Move-only* - Moves ownership of the pointer. The wrapped object remains un-moved in its heap allocation.
+* *Not Comparable* - Doesn't provide `operator==`
+
+```c++
+// <unifex/any_unique.hpp>
+namespace unifex
+{
+  template<typename... CPOs>
+  class any_unique {
+  public:
+    // Decay-copies the argument. Uses std::allocator to heap-allocate storage.
+    template<typename T>
+      requires std::constructible_from<std::remove_cvref_t<T>, T> &&
+          !(std::same_as<remove_cvref_t<T>, any_unique> || instance-of-in_place_type_t<T>>)
+    /* implicit */ any_unique(T&& x);
```

Comment:
I'm pretty sure the actual implementation constrains on this, I just omitted it from the documentation here because it made the requires clause overly complicated.
I can add it back here if you think it's important.

Reply:
Actually, it seems like the current `any_unique` constructor doesn't actually constrain on the type being type-erasable...